app/models/location.py

```diff
@@ -4,10 +4,42 @@
 class ZIPCode(db.Model):
     __tablename__ = 'zip_codes'
     id = db.Column(db.Integer, primary_key=True)
-    zip_code = db.Column(db.String(5), unique=True)
+    zip_code = db.Column(db.String(5), unique=True, index=True)
     users = db.relationship('User', backref='zip_code', lazy='dynamic')
     addresses = db.relationship('Address', backref='zip_code', lazy='dynamic')
 
+    def __init__(self, zip_code):
```

Comment:
Should we leave a comment saying that this constructor should never be explicitly called by a user of the model?